spec/openapi3.yaml

```diff
@@ -840,6 +841,205 @@ paths:
+          description: invalid argument
+  "/account/{accountId}/metadata/{key}":
+    get:
+      tags:
+      - Metadata routes
+      summary: Get account metadata
+      description: Returns the account metadata given an account id and a key
+      operationId: getAccountMetadataByKey
+      parameters:
+        - $ref: "#/components/parameters/accountId"
+        - $ref: "#/components/parameters/metadataKey"
+      responses:
+        "200":
+          description: success
+          content:
+            application/json:
+              schema:
+                  $ref: "#/components/schemas/MetadataEntriesDTO"
+        "404":
+          description: resource not found
+        "409":
+          description: invalid argument
+  "/mosaic/{mosaicId}/metadata/{key}":
+    get:
+      tags:
+      - Metadata routes
+      summary: Get mosaic metadata
+      description: Returns the mosaic metadata given a mosaic id and a key
+      operationId: getMosaicMetadataByKey
+      parameters:
+        - $ref: "#/components/parameters/mosaicId"
+        - $ref: "#/components/parameters/metadataKey"
+      responses:
+        "200":
+          description: success
+          content:
+            application/json:
+              schema:
+                  $ref: "#/components/schemas/MetadataEntriesDTO"
+        "404":
+          description: resource not found
+        "409":
+          description: invalid argument
+  "/namespace/{namespaceId}/metadata/{key}":
+    get:
+      tags:
+      - Metadata routes
+      summary: Get namespace metadata
+      description: Returns the namespace metadata given a namespace id and a key
+      operationId: getNamespaceMetadataByKey
+      parameters:
+        - $ref: "#/components/parameters/namespaceId"
+        - $ref: "#/components/parameters/metadataKey"
+      responses:
+        "200":
+          description: success
+          content:
+            application/json:
+              schema:
+                  $ref: "#/components/schemas/MetadataEntriesDTO"
+        "404":
+          description: resource not found
+        "409":
+          description: invalid argument
+  "/account/{accountId}/metadata/{key}/signer/{publicKey}":
+    get:
+      tags:
+      - Metadata routes
+      summary: Get account metadata
+      description: Returns the account metadata given an account id, a key, and a signer
+      operationId: getAccountMetadataByKeyAndSigner
+      parameters:
+        - $ref: "#/components/parameters/accountId"
+        - $ref: "#/components/parameters/metadataKey"
+        - $ref: "#/components/parameters/publicKey"
+      responses:
+        "200":
+          description: success
+          content:
+            application/json:
+              schema:
+                  $ref: "#/components/schemas/MetadataDTO"
+        "404":
+          description: resource not found
+        "409":
+          description: invalid argument
+  "/mosaic/{mosaicId}/metadata/{key}/signer/{publicKey}":
```

Comment:
`sender` instead of 'signer`? to match the `senderPublicKey`

Reply:
@dgarcia360 this was your call

Reply:
Going to use `signer` for now as time limited. if needed can change it later one. Concern from is that the signer could potentially be confusing if the tx get signed by a cosigner. But will make it clear in method doc 👍

Reply:
@rg911 I agree that ``sender`` fits better, well seen. 

@Vektrat Yes, the endpoints were suggested before the MongoDB revision. Could we apply this change before merging the metadata PR in REST? Thanks

Reply:
no problem,changed it

Reply:
Need to change the method name as well